torch/distributed/distributed_c10d.py

```diff
@@ -2034,15 +2034,23 @@ def scatter_object_list(
     scatter_object_output_list[0] = _tensor_to_object(output_tensor, obj_tensor_size)
 
 
-def all_gather(tensor_list, tensor, group=None, async_op=False):
+def all_gather(output, tensor, group=None, async_op=False):
```

Comment:
IMHO this looks backward compatibility breaking and I'm wondering if we could just make `_all_gather_base` public.

Reply:
Hi @crcrpar, I started writing my reply to @rohan-varma on why the unification and why such signature before seeing your comment. Do you mind checking if that responds to the second part of your question? Please feel free to let me know if you have any comment about that thought process.

Regarding backward compatibility, I will think more about it.